File: js/task3b/S3_FN_06.js
function mapValues(obj, fn) {
  const result = {};
  for (const key in obj) {
    if (obj.hasOwnProperty(key)) { 
      result[key] = fn(obj[key], key, obj);
    }
  }

  return result;
}
const original = { a: 1, b: 2, c: 3 };
const squared = mapValues(original, x => x * x);

console.log(squared);
console.log(original);

File: python/task3b/S3_FN_06.py
def map_values(obj, fn):
  result = {}
  for key, value in obj.items():
      result[key] = fn(value, key, obj)
  return result


original = {"a": 1, "b": 2, "c": 3}
squared = map_values(original, lambda x, k, o: x * x)

print(squared)
print(original)
